src/runtime_src/core/tools/xbtracer/src/app/launcher.cpp

```diff
@@ -0,0 +1,534 @@
+ * It terminates when there are no more values to process.
+ */
+template <typename T, typename... Args>
+void log_format(std::ostringstream& oss, const T& t, const Args&... args)
+{
+  /* Warning: implicitly decay an array into a pointer
+  * The function is explicitly designed to handle any type, including arrays
+  * Behavior of array-to-pointer decay as safe in this context
+  */
+  oss << t; /* NOLINT(cppcoreguidelines-pro-bounds-array-to-pointer-decay,
+  hicpp-no-array-decay) */
+  log_format(oss, args...);
+}
+
+/*
+ * Function to trace the error log
+ */
+template <typename... Args>
+void log_e(const Args&... args)
+{
+  std::ostringstream oss;
+  oss << "[" << launcher::getInstance().m_name.data() << "] E: ";
+  log_format(oss, args...);
+  std::cout << oss.str() << std::endl;
+}
+
+/*
+ * Function to trace the fatal log
+ */
+template <typename... Args>
+void log_f(const Args&... args)
+{
+  std::ostringstream oss;
+  oss << "[" << launcher::getInstance().m_name.data() << "] F: ";
+  log_format(oss, args...);
+  throw std::runtime_error(oss.str() + ". Aborted!\n");
+}
+
+/*
+ * Function to trace the debug log
+ */
+template <typename... Args>
+void log_d(const Args&... args)
+{
+  if (launcher::getInstance().m_debug) {
+    std::ostringstream oss;
+    oss << "[" << launcher::getInstance().m_name.data() << "] D: ";
+    log_format(oss, args...);
+    std::cout << oss.str() << std::endl;
+  }
+}
+
+/*
+ * thread safe wrapper of putenv()
+ */
+void putenv_t(std::string& new_entry)
+{
+  std::lock_guard<std::mutex> lock(env_mutex);
+  #ifdef _WIN32
+  _putenv(new_entry.data());
+  #else
+  // NOLINTNEXTLINE(concurrency-mt-unsafe) - protected by env_mutex
+  putenv(new_entry.data());
+  #endif
+}
+
+#ifdef _WIN32
+/*
+ * Check the architectural compatability(32bit vs 64bit) between the app and
+ * lib.
+ */
+static bool check_compatibility(HANDLE parent, HANDLE child)
+{
+  static launcher& app = launcher::getInstance();
+  BOOL is_parent_wow64 = FALSE, is_child_wow64 = FALSE;
+
+  IsWow64Process(parent, &is_parent_wow64);
+  IsWow64Process(child, &is_child_wow64);
+
+  if (is_parent_wow64 != is_child_wow64) {
+    log_e(app.m_name.data(), "is", (is_parent_wow64 ? w64 : w32),
+          "-bit but target application is ", (is_child_wow64 ? w64 : w32),
+          "-bit");
+    return false;
+  }
+
+  return true;
+}
+
+// Function to set an environment variable
+bool SetEnvVariable(const std::string& name, const std::string& value)
+{
+  LPCSTR lpName = name.c_str();
+  LPCSTR lpValue = value.c_str();
+
+  BOOL result = SetEnvironmentVariableA(lpName, lpValue);
+
+  if (result == 0) {
+    DWORD errorCode = GetLastError();
+    std::cerr << "Failed to set environment variable. Error code: " << errorCode
+              << std::endl;
+    return false;
+  }
+
+  return true;
+}
+
+/*
+ * Function to get the current precise system time as pair for string and
+ * formated string
+ */
+std::pair<std::string, std::string> GetCurrentTimeAsString()
```

Comment:
I have, what some think, a weird style that prefers keeping as much information to the left margin as possible.  IMO it makes scanning code much easier when all functions names are aligned left as opposed to right after complex return types.  E.g.
```suggestion
std::pair<std::string, std::string> 
GetCurrentTimeAsString()
```
I am not adamant about this style except for code that I maintain.   It's up to you how you style your own files.  Just wanted to mention it.

Reply:
Adopted, please let me know if find any inconsistency.